api/src/test/java/org/pmiops/workbench/api/ClusterControllerTest.java

```diff
@@ -242,7 +294,96 @@ public void testListClusters() throws Exception {
   }
 
   @Test
-  public void testListClustersUnknownStatus() throws Exception {
+  public void testDeleteClustersInProject() {
+    List<ListClusterResponse> listClusterResponseList = ImmutableList.of(testFcClusterListResponse);
+    when(notebookService.listClustersByProjectAsAdmin(BILLING_PROJECT_ID))
+        .thenReturn(listClusterResponseList);
+
+    clusterController.deleteClustersInProject(
+        BILLING_PROJECT_ID,
+        new ListClusterDeleteRequest()
+            .clustersToDelete(ImmutableList.of(testFcCluster.getClusterName())));
+    verify(notebookService)
+        .deleteClusterAsAdmin(BILLING_PROJECT_ID, testFcCluster.getClusterName());
+    verify(clusterAuditor)
+        .fireDeleteClustersInProject(
+            BILLING_PROJECT_ID,
+            listClusterResponseList.stream()
+                .map(ListClusterResponse::getClusterName)
+                .collect(Collectors.toList()));
+  }
+
+  @Test
+  public void testDeleteClustersInProjectDeleteSome() {
+    List<ListClusterResponse> listClusterResponseList =
+        ImmutableList.of(testFcClusterListResponse, testFcClusterListResponse2);
+    List<String> clustersToDelete = ImmutableList.of(testFcCluster.getClusterName());
+    when(notebookService.listClustersByProjectAsAdmin(BILLING_PROJECT_ID))
+        .thenReturn(listClusterResponseList);
+
+    clusterController.deleteClustersInProject(
+        BILLING_PROJECT_ID, new ListClusterDeleteRequest().clustersToDelete(clustersToDelete));
+    verify(notebookService, times(clustersToDelete.size()))
+        .deleteClusterAsAdmin(BILLING_PROJECT_ID, testFcCluster.getClusterName());
+    verify(clusterAuditor, times(1))
+        .fireDeleteClustersInProject(BILLING_PROJECT_ID, clustersToDelete);
+  }
+
+  @Test
+  public void testDeleteClustersInProjectDeleteDoesNotAffectOtherProjects() {
+    List<ListClusterResponse> listClusterResponseList =
+        ImmutableList.of(testFcClusterListResponse, testFcClusterListResponse2);
+    List<String> clustersToDelete =
+        ImmutableList.of(testFcClusterDifferentProject.getClusterName());
+    when(notebookService.listClustersByProjectAsAdmin(BILLING_PROJECT_ID))
+        .thenReturn(listClusterResponseList);
+
+    clusterController.deleteClustersInProject(
+        BILLING_PROJECT_ID, new ListClusterDeleteRequest().clustersToDelete(clustersToDelete));
+    verify(notebookService, times(0))
+        .deleteClusterAsAdmin(BILLING_PROJECT_ID, testFcCluster.getClusterName());
+    verify(clusterAuditor, times(0))
+        .fireDeleteClustersInProject(BILLING_PROJECT_ID, clustersToDelete);
+  }
+
+  @Test
+  public void testDeleteClustersInProjectNoClusters() {
```

Comment:
This is something I was curious about: do we want a missing param (null) to have different semantics from an empty-list param?  We should be explicit about this in the swagger, if so.

Reply:
I think we do, because I think not providing a list should give all, but an empty list should behave the same way as any list. I can be more explicit in swagger.

Reply:
Basically I think there should be some default that says include all clusters, and the body parameter can't be null. We could also do this with a bool on the object that says 'deleteAll' or something, but that seems like extra that we don't need.